tests/ui/map_identity.rs

```diff
@@ -20,6 +20,27 @@ fn main() {
     });
     let _: Result<u32, u32> = Ok(1).map_err(|a| a);
     let _: Result<u32, u32> = Ok(1).map_err(|a: u32| a * 42);
+    issue7189();
+}
+
+fn issue7189() {
+    // should lint
+    let x = [(1, 2), (3, 4)];
+    let _ = x.iter().map(|(x, y)| (x, y));
+    let _ = x.iter().map(|(x, y)| {
+        return (x, y);
+    });
+    let _ = x.iter().map(|(x, y)| return (x, y));
+
+    let y = [(1, 2, (3, (4,))), (5, 6, (7, (8,)))];
+    let _ = y.iter().map(|(x, y, (z, (w,)))| (x, y, (z, (w,))));
+
+    // should not lint
+    let _ = x.iter().map(|(x, y)| (x, y, y));
+    let _ = x.iter().map(|(x, _y)| (x,));
+    let _ = x.iter().map(|(x, _)| (x,));
+    let _ = x.iter().map(|(x, ..)| (x,));
+    let _ = y.iter().map(|(x, y, (z, _))| (x, y, (z, z)));
```

Comment:
Oh I *think* I see what you were trying to get at, maybe..? Should it *not* lint if there are type annotations? I just came up with this case that would break if the last map was removed, because it provides it with type hints, and without them type inference breaks:
```rs
use std::future::{pending, Pending};

let _ = [1].iter().map(|_| pending()).map(|x: Pending<i32>| x);
```

Reply:
Exactly that.